### PR TITLE
Time series plot

### PR DESCRIPTION
**Pull request by zouwenjiao**
Things still needs to do

- [x] check inputs

- [x] Add tests

- [x] Fix the bug "package or namespace load failed for ‘forecast’ in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]):
 there is no package called ‘nnet’"

### CODE REVIEW

File: R/ts_plot.R [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+ts_plot <- function(data, col, frequency){
+  df <- readr::read_csv(data)
+  ts <- ts(df[[col]], frequency = frequency)
+
+  if (time(ts)[2] - time(ts)[1] != time(ts)[3] - time(ts)[2]){
+    stop("This time series is irregular.")
+  }
+
+  if (time(ts)[2] - time(ts)[1] != 1/frequency){
+    stop("The frequency is wrong.")
```
Comment (tejasph) [inline R/ts_plot.R]: You could potentially combine these into the same if/else statement, although it is fine the way it is as well. Maybe add more descriptive error messages to help guide the user in debugging.

File: R/ts_plot.R [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
+  }
+
+  if (frequency == 1){
+    plt <- forecast::autoplot(ts, main = "Time series data", xlab="Time", ylab = col)
+  }
+  else {
+    decomposed <- stats::decompose(ts)
+    plt <- forecast::autoplot(decomposed, main = "Decomposed Time series data", xlab="Time", ylab = col)
```
Comment (tejasph) [inline R/ts_plot.R]:
```suggestion
    plt <- forecast::autoplot(decomposed, main = "Decomposed Time-Series Data", xlab="Time", ylab = col)
```

File: R/ts_plot.R [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
 #' @param frequency int: the desired time interval
 #'
-#' @return a figure with 4 subplots
+#' @return ggplot Object
 #' @export
 #' @examples
-#' ts_plot("https://raw.githubusercontent.com/plotly/datasets/master/timeseries.csv", A, irregular, 3)
-ts_plot <- function(data, col, type, frequency){
-  print("hello")
+#' ts_plot("~/Desktop/ts4_jjshares.csv", "earnings", 4)
```
Comment (tejasph) [inline R/ts_plot.R:14]: Maybe replace with a built in dataset so that any user could replicate this example

**Review by tejasph — CHANGES_REQUESTED** [review]
Really great job Wenjiao. Just a few minor comments to look over.

File: R/ts_plot.R (hunk 1 above)
Comment (zouwenjiao) [reply R/ts_plot.R]: OK, I have added more descriptive error messages to it.

File: R/ts_plot.R (hunk 3 above)
Comment (zouwenjiao) [reply R/ts_plot.R:14]: Since it is the path of  a csv file, it is hard to use a built-in dataset...

**Review by tejasph — APPROVED** [review]

**Pull request merged**